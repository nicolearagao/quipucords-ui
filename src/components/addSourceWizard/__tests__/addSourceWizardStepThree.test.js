import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { ConnectedAddSourceWizardStepThree, AddSourceWizardStepThree } from '../addSourceWizardStepThree';

describe('AccountWizardStepResults Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a basic component with error', () => {
    const store = generateEmptyStore({
      addSourceWizard: {
        error: true,
        errorMessage: 'Lorem ipsum'
      }
    });
    const component = renderComponent(
      <Provider store={store}>
        (<ConnectedAddSourceWizardStepThree />
      </Provider>
    );

    expect(component).toMatchSnapshot('basic');
  });

  it('should render a wizard results step with error', async () => {
    const props = {
      add: false,
      error: true,
      errorMessage: 'lorem ipsum'
    };

    const component = await shallowComponent(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('error updated');

    const componentErrCreated = await component.setProps({ add: true });
    expect(componentErrCreated).toMatchSnapshot('error created');
  });

  it('should render a wizard results step with pending', async () => {
    const props = {
      add: false,
      pending: true,
      name: 'Dolor'
    };

    const component = await shallowComponent(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('pending updated');

    const componentPendingCreated = await component.setProps({ add: true });
    expect(componentPendingCreated).toMatchSnapshot('pending created');
  });

  it('should render a wizard results step with fulfilled', async () => {
    const props = {
      add: false,
      fulfilled: true,
      name: 'Dolor'
    };

    const component = await shallowComponent(<AddSourceWizardStepThree {...props} />);
    expect(component).toMatchSnapshot('fulfilled updated');

    const componentFulfilledCreated = await component.setProps({ add: true });
    expect(componentFulfilledCreated).toMatchSnapshot('fulfilled created');
  });
});
