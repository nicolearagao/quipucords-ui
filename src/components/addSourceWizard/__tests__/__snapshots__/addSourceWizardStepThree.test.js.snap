// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountWizardStepResults Component should render a basic component with error: basic 1`] = `
<provider
  props="{
  "store": {},
  "children": [
    {},
    {
      "displayName": "Connect(AddSourceWizardStepThree)",
      "props": {},
      "type": {
        "compare": null
      }
    }
  ]
}"
>
  (
  <div
    class="pf-c-empty-state quipucords-empty-state"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-hidden="true"
        class="pf-c-empty-state__icon"
        fill="#c9190b"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        />
      </svg>
      <h3
        class="pf-c-title pf-m-lg"
        data-ouia-component-id="OUIA-Generated-Title-1"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        t(form-dialog.empty-state_title_add-source_error, {"context":"edit","name":null})
      </h3>
      <div
        class="pf-c-empty-state__body"
      >
        t(form-dialog.empty-state_description_add-source_error, {"context":" ","name":null})
      </div>
    </div>
  </div>
</provider>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with error: error created 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    color="#c9190b"
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source, {"context":"error","name":null})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source_error, {"context":" ","name":null})
  </EmptyStateBody>
</EmptyState>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with error: error updated 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    color="#c9190b"
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source_error, {"context":"edit","name":null})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source_error, {"context":" ","name":null})
  </EmptyStateBody>
</EmptyState>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with fulfilled: fulfilled created 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    color="#3e8635"
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source, {"context":"","name":"Dolor"})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source, {"context":" ","name":"Dolor"})
  </EmptyStateBody>
</EmptyState>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with fulfilled: fulfilled updated 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    color="#3e8635"
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source, {"context":"edit","name":"Dolor"})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source, {"context":" ","name":"Dolor"})
  </EmptyStateBody>
</EmptyState>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with pending: pending created 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source, {"context":"pending","name":"Dolor"})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source_pending, {"context":" ","name":"Dolor"})
  </EmptyStateBody>
</EmptyState>
`;

exports[`AccountWizardStepResults Component should render a wizard results step with pending: pending updated 1`] = `
<EmptyState
  className="quipucords-empty-state"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h3"
  >
    t(form-dialog.empty-state_title_add-source_pending, {"context":"edit","name":"Dolor"})
  </Title>
  <EmptyStateBody>
    t(form-dialog.empty-state_description_add-source_pending, {"context":"edit","name":"Dolor"})
  </EmptyStateBody>
</EmptyState>
`;
